### ISSUE TITLE
Integrate fixes from Alfresco patch

### ISSUE THREAD

**Issue by marcelmay** [greenmail-mail-test/greenmail#37]
Alfresco patched GreenMail for various issues.

Evaluate and reintegrate the ~~[patch](https://github.com/Alfresco/community-edition/blob/master/projects/3rd-party/src/greenmail-1.3-src.diff)~~ [patch](https://github.com/Alfresco/community-edition/commit/60c49f6f7c65b084be9eb7ff38f89863231b9640#diff-98db6ae3b64864b91c01d139f197cc95).

**Comment by markrogersalfresco** [comment]
If you need more details of why we have patched what we have then it will be in Alfresco's JIRA.    I can probably give details.    We should also have integration tests that could be adapted. 

I'll start by contributing an easy fix just to get the ball rolling.

**Comment by buildscientist** [comment]
@markrogersalfresco Hi - do you have any tests that go along with the patches you mentioned in #46? 

Thanks, 
Youssuf

**Comment by vguna** [comment]
Using 1.5.1 here facing the problem described here (encodings):

https://github.com/greenmail-mail-test/greenmail/issues/32

Body contains german umlauts. Using `GreenMailUtil.getBody(mimeMessage)` shows raw encodings like 
`Bitte best=C3=A4tigen Sie die Registrierung durch einen Klick auf u.a. Link=.`

Currently working around using Spring's `MimeMessageHelper` like that:

```java
		MimeMessageHelper helper = new MimeMessageHelper(mimeMessage);
		String body = helper.getMimeMessage().getContent();
```

Returns `Bitte bestätigen Sie die Registrierung durch einen Klick auf u.a. Link.`

**Comment by marcelmay** [comment]
@vguna , why not use plain JavaMail for getting the content:

    mimeMessage.getContent()

MimeMessageHelper#getMimeMessage() directly returns the wrapped mime message,
so you could leave out the MMH wrapper.

I'll look into GMU.getBody, looks like it is returning the raw mail content.

You can check out the [test case for umlauts](https://github.com/greenmail-mail-test/greenmail/blob/master/greenmail-core/src/test/java/com/icegreen/greenmail/test/ImapServerTest.java#L48).

**Comment by vguna** [comment]
Yeah, you're right :). Interesting why the encoding isn't working for me using GMU. I'll have to recheck...
